File: api/src/routes/notification.js
const express = require("express");

const router = express.Router();

const {
  create,
  update,
  remove,
  getById,
  getAll,
} = require("../controllers/notification");

router.post("/notification", create);
router.delete("/notification/:id", remove);
router.put("/notification/:id", update);
router.get("/notification", getAll);
router.get("/notification/:id", getById);

module.exports = router;

File: client/lib/controllers/notification.dart
import 'package:client/config/core.dart';
import 'package:client/models/notification.dart';

class NotificationController {
  static Future<Notification?> getById(String nId) async {
    try {
      final response = await http!.get('notification/$nId');
      return Notification.fromMap(response.data['datas']);
    } catch (err) {
      print(err);
      return null;
    }
  }

  static Future<List<Notification>?> getAll() async {
    try {
      final response = await http!.get('notification');
      return (response.data['datas'] as List)
          .map((e) => Notification.fromMap(e))
          .toList();
    } catch (err) {
      print(err);
      return null;
    }
  }

  static Future<Notification?> create(Notification notification) async {
    try {
      final data = notification.asMap();
      final response = await http!.post('notification', data: data);
      return Notification.fromMap(response.data['datas']);
    } catch (err) {
      print(err);
      return null;
    }
  }
}
